### PR TITLE
methods-modular_programs-easy-q6

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/methods/modular_programs/easy/q6/Century.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package methods.modular_programs.easy.q6;
+
+/*
+Write a Java Program that asks users for the year that they were born in, then returns what century they were born in.
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/Century.java:4]: Change question text to:
`
Write a Java Program that asks users for the year they were born, then returns what century that year is in. The program should have a method called centuryCalc to determine this, which works for any century from the 1500's to 2000's. If the input is in any other century, return "unknown century" .
`

File: src/methods/modular_programs/easy/q6/Century.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class Century
+{
+    public static void main( String[] args )
+    {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Please enter the year you were born in: ");
+        int userYear = input.nextInt();
+        System.out.println("You're from the " + centuryCalc(userYear));
+    }
+    public static String centuryCalc(int userYear)
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/Century.java:22]: I think we should update this question to work with more centuries (see question text).

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q6/Century.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package methods.modular_programs.easy.q6;
+
+/*
+Write a Java Program that asks users for the year they were born, then returns what century that year is in. The program
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/Century.java:4]: Change the question text to:
`Write a Java Program that asks users for the year they were born, then prints the century the individual was born. The program should have a method called centuryCalc, which determines the century and works for any century from the 1500s to the 2000's. If the input is in any other century, return "Unknown Century".
`

Comment (Hedgemon4) [reply src/methods/modular_programs/easy/q6/Century.java:4]: We should also maybe add a note that the 1500s is the 16th century and so on

File: src/methods/modular_programs/easy/q6/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                Arguments.of("1634", "17th Century"),
+                Arguments.of("1555", "16th Century"),
+                Arguments.of("1265", "Unknown Century"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("centuryCalcInputProvider")
+    void correctCenturyCalcMethod(int in, String century) throws Throwable {
+        Object output = MethodUtil.invokeIfMethodExists(Century.class, "centuryCalc", new Object[]{in}, int.class);
+        CustomAssertions._assertEquals(century, output, "Your centuryCalc method does not return the right century.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/MainTest.java:77]:
```suggestion
        CustomAssertions._assertEquals(century, output, "Your centuryCalc method does not return the correct century.");
```

File: src/methods/modular_programs/easy/q6/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                Arguments.of("1782", "18th Century"),
+                Arguments.of("1634", "17th Century"),
+                Arguments.of("1555", "16th Century"),
+                Arguments.of("1265", "Unknown Century"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("centuryCalcInputProvider")
+    void correctCenturyCalcMethod(int in, String century) throws Throwable {
+        Object output = MethodUtil.invokeIfMethodExists(Century.class, "centuryCalc", new Object[]{in}, int.class);
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/MainTest.java:76]: Change to methodTest

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q6/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                {in, int.class}
+        };
+        MethodTest m = new MethodTest(Century.class, "centuryCalc", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(century, output, "Your centuryCalc method does not return the correct century.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void printsOutputCorrectly(String in, String century) throws InvalidClauseException {
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/MainTest.java:86]: Again, if we are providing an int as input, I think we should provide the method with int and then String.valueOf(x) in the method

File: src/methods/modular_programs/easy/q6/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                Arguments.of(1500, "16th Century"),
+                Arguments.of(1499, "Unknown Century"),
+                Arguments.of(1498, "Unknown Century"),
+                Arguments.of(567, "Unknown Century"),
+                Arguments.of(-5, "Unknown Century"));
+    }
+
+    static Stream<Arguments> mainMethodInputProvider() {
+        return Stream.of(
+                Arguments.of("2022", "21st Century"),
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q6/MainTest.java:64]: Change to provide int instead.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]
